cli/src/diff_util.rs

```diff
@@ -402,50 +402,49 @@ pub fn show_color_words_diff(
                     show_color_words_diff_hunks(&[], &right_content, formatter)?;
                 }
             } else if right_value.is_present() {
-                let left_content = diff_content(repo, &path, &left_value)?;
-                let right_content = diff_content(repo, &path, &right_value)?;
-                let description = match (left_value.into_resolved(), right_value.into_resolved()) {
+                let description = match (left_value.as_resolved(), right_value.as_resolved()) {
                     (
-                        Ok(Some(TreeValue::File {
+                        Some(Some(TreeValue::File {
                             executable: left_executable,
                             ..
                         })),
-                        Ok(Some(TreeValue::File {
+                        Some(Some(TreeValue::File {
                             executable: right_executable,
                             ..
                         })),
                     ) => {
-                        if left_executable && right_executable {
+                        if *left_executable && *right_executable {
                             "Modified executable file".to_string()
-                        } else if left_executable {
+                        } else if *left_executable {
                             "Executable file became non-executable at".to_string()
-                        } else if right_executable {
+                        } else if *right_executable {
                             "Non-executable file became executable at".to_string()
                         } else {
                             "Modified regular file".to_string()
                         }
                     }
-                    (Err(_), Err(_)) => "Modified conflict in".to_string(),
-                    (Err(_), _) => "Resolved conflict in".to_string(),
-                    (_, Err(_)) => "Created conflict in".to_string(),
-                    (Ok(Some(TreeValue::Symlink(_))), Ok(Some(TreeValue::Symlink(_)))) => {
+                    (None, None) => "Modified conflict in".to_string(),
+                    (None, _) => "Resolved conflict in".to_string(),
+                    (_, None) => "Created conflict in".to_string(),
+                    (Some(Some(TreeValue::Symlink(_))), Some(Some(TreeValue::Symlink(_)))) => {
                         "Symlink target changed at".to_string()
                     }
-                    (Ok(left_value), Ok(right_value)) => {
-                        let left_type = basic_diff_file_type(&Merge::resolved(left_value));
-                        let right_type = basic_diff_file_type(&Merge::resolved(right_value));
+                    (Some(left_value), Some(right_value)) => {
+                        let left_type = basic_diff_file_type(&Merge::resolved(left_value.clone()));
+                        let right_type =
+                            basic_diff_file_type(&Merge::resolved(right_value.clone()));
```

Comment:
nit: match by `(Some(_), Some(_))` and pass the original `&left_value`/`&right_value`.

Reply:
Good idea!